aptos-move/e2e-move-tests/src/tests/access_path_test.rs

```diff
@@ -30,6 +31,8 @@ fn access_path_panic() {
 
     let cm = CompiledModule {
         version: 6,
+        compiler_version_major: COMPILER_VERSION_MAJOR_1,
```

Comment:
I thought a bit about it and I think we should probably just make it a string. This gives as then the flexibility to evolve versioning schemes outside of the file format.

For example, there can be other compilers (from different languages than Move) which generate Move bytecode. Then we need to also encode the compiler name. We can develop such versioning schema later.

For now what we just want to do is disallow-list certain compilers. So for the v2 compiler we set it to a string "Move on Aptos Compiler v1.0" (we can fix detail later). Any bytecode with this version tag will be disallowed on mainnet. The thing is also that different than the bytecode version, all the different compilers are compatible. Only thing we want is block certain compilers.

Reply:
Sure, for this field, should we still use this on the serialization part to just control the encoding of V7 like what we do here?

Reply:
This field is completely unrelated to the control of serialization. For that we have already the file format version.

It is just a tag and has no semantic meaning for the file format. Maybe we should call it `compiler_id: String` to avoid any confusion.

Reply:
Eventually, what we want to have on chain is a list of 'approved' compiler ids. This list can change from network to network. We can also have a wildcard allowing anything. Basically `approved_compiler_ids: Option<vector<String>>`. This need to be stored in an on-chain config, a feature flag is not enough for this.